src/language_features/code_lens.rs

```diff
@@ -32,10 +32,54 @@ pub fn text_document_code_lens(meta: EditorMeta, ctx: &mut Context) {
         partial_result_params: PartialResultParams::default(),
     };
     ctx.call::<CodeLensRequest, _>(meta, req_params, |ctx: &mut Context, meta, result| {
-        editor_code_lens(meta, result, ctx)
+        editor_code_lens(meta.clone(), result.clone(), ctx);
+        inlay_hints_from_editor_code_lens(meta, result.unwrap_or_default(), ctx)
     });
 }
 
+fn inlay_hints_from_editor_code_lens(
+    meta: EditorMeta,
+    mut lenses: Vec<CodeLens>,
+    ctx: &mut Context,
+) {
+    let document = match ctx.documents.get(&meta.buffile) {
+        Some(document) => document,
+        None => {
+            return;
+        }
+    };
+    lenses.sort_by_key(|lens| lens.range.start);
+
+    let inlay_hints_from_lenses = lenses
+        .iter()
+        .map(|lens| {
+            let label = lens.command.as_ref().map_or("", |v| &v.title);
+            if label.trim().len() > 0 {
```

Comment:
clippy demands we use `is_empty()` but actually I think it's better to use a `filter()` before the `map()`

TBH I think an empty title is probably a bug in the language server, and I don't expect that special-casing it is useful. Happy to proven wrong of course, is there any server that sends empty titles?

Of course we do want to filter away cases where the command is not set (because it's not resolved yet; we only resolve upon user action). Below we do this with

        .filter(|lens| lens.command.is_none())

so the most obvious code here would be

        .filter(|lens| lens.command.is_some())

Reply:
Removed the whole `if label.trim().len() > 0` business. Assumption: if there is a code lens sent by the server, there would always be a title. Simplifies the logic and I can't think of scenarios where we would have a title but not a code lens.